src/category_theory/triangulated/rotate.lean

```diff
@@ -0,0 +1,629 @@
+    dsimp,
+    ext,
+    {
+      simp only [triangle_morphism.comp_trimor1,
+      triangle_morphism.inv_rotate_trimor1, triangle_morphism.rotate_trimor3],
+      dsimp,
+      simp only [iso.hom_inv_id_app, assoc, equivalence.inv_fun_map,
+      nat_iso.cancel_nat_iso_inv_left],
+      dsimp,
+      simp only [comp_id],
+    },
+    {
+      simp only [triangle_morphism.comp_trimor2,
+      triangle_morphism.inv_rotate_trimor2, triangle_morphism.rotate_trimor1,
+      comp_id f.trimor2, id_comp f.trimor2],
+    },
+    {
+      simp only [triangle_morphism.comp_trimor3, triangle_morphism.rotate_trimor2,
+      triangle_morphism.inv_rotate_trimor3, comp_id f.trimor3, id_comp f.trimor3],
+    },
+  end
+}
+
+/--
+The natural transformations between the identity functor on triangles and the composition
+of a rotation with an inverse rotation are natural isomorphisms (they are isomorphisms in the
+category of functors).
+-/
+@[simps]
+def rot_comp_inv_rot :𝟭 (triangle C) ≅ (rotate C) ⋙ (inv_rotate C) :=
+{
+  hom := rot_comp_inv_rot_hom C,
+  inv := rot_comp_inv_rot_inv C,
+  hom_inv_id' := begin
+    ext T,
+    {
+      simp only [functor.id_obj, congr_arg_mpr_hom_left, triangle_morphism.comp_trimor1,
+      triangle_category_to_category_struct_comp, rot_comp_inv_rot_inv_app,
+      rot_comp_inv_rot_hom_app, iso.hom_inv_id_app, triangle_morphism.id_comp,
+      nat_trans.id_app, triangle_category_to_category_struct_id,
+      triangle_morphism_id_trimor1, eq_to_hom_refl, congr_arg_mpr_hom_right,
+      triangle_morphism.comp_id, functor.comp_obj, nat_trans.comp_app],
+      dsimp,
+      refl,
+    },
+    {
+      simp only [functor.id_obj, congr_arg_mpr_hom_left, triangle_category_to_category_struct_comp,
+      rot_comp_inv_rot_inv_app, triangle_morphism.comp_trimor2, rot_comp_inv_rot_hom_app,
+      triangle_morphism.id_comp, nat_trans.id_app, triangle_category_to_category_struct_id,
+      triangle_morphism_id_trimor2, eq_to_hom_refl, congr_arg_mpr_hom_right,
+      triangle_morphism.comp_id, functor.comp_obj, nat_trans.comp_app],
+      dsimp,
+      simp only [comp_id],
+    },
+    {
+      simp only [functor.id_obj, congr_arg_mpr_hom_left, triangle_category_to_category_struct_comp,
+      rot_comp_inv_rot_inv_app, triangle_morphism.comp_trimor3, rot_comp_inv_rot_hom_app,
+      triangle_morphism.id_comp, nat_trans.id_app, triangle_category_to_category_struct_id,
+      triangle_morphism_id_trimor3, eq_to_hom_refl, congr_arg_mpr_hom_right,
+      triangle_morphism.comp_id, functor.comp_obj, nat_trans.comp_app],
+      dsimp,
+      simp only [comp_id],
+    }
+  end,
+  inv_hom_id' := begin
+    ext T,
+    {
+      simp,
+      dsimp,
+      refl,
+    },
+    {
+      simp,
+      dsimp,
+      simp only [comp_id],
+    },
+    {
+      simp,
+      dsimp,
+      simp only [comp_id],
+    }
+  end -- (deterministic) timeout when replace simp with squeeze_simp
```

Comment:
Have you tried sorrying out all the other fields, while you try `squeeze_simp`?

In any case, these `simp only ...` proofs are pretty gross, it would be nice if we could avoid them. :-)

Reply:
I can use the sorry method to swap the "simp"s for "simp only"s, but if another method is preferred, I can have a go at that instead. How would you suggest going about these proofs instead?

Reply:
Just leave them all out. They are not needed, as `obviously` will do them for you. In fact, I think you can remove every single proof in this file, with the exceptions of

* `rotate.comm₃'`
* `inv_rotate.comm₁'`
* `rot_comp_inv_rot_hom.comm₃'`

Once we've got down to that point, we can do some further polishing to see if we can reasonably get rid of them as well. :-)

Reply:
Have removed proofs that are "obvious". Are there any further changes you would suggest for the proofs you mentioned?